internal/pkg/cli/task_run.go

```diff
@@ -501,6 +516,30 @@ func (o *runTaskOpts) confirmSecretsAccess() error {
 	return nil
 }
 
+func (o *runTaskOpts) validateEnvCompatibility(app, env string) error {
```

Comment:
Given that the purpose of this function is only to validate the version when it's specifically job - 

```suggestion
func (o *runTaskOpts) validateEnvCompatibilityForGenerateJobCmd(app, env string) error {
```